node_helper.js

```diff
@@ -1,23 +1,42 @@
 var dvb = require('dvbjs');
 var NodeHelper = require("node_helper");
+const Log = require("logger");
 
 module.exports = NodeHelper.create({
 
+    start: function() {
+        Log.log("Starting node helper for: " + this.name);
+        this.stop = {};
+    },
+    monitor: function(payload) {
+        var self = this;
+        dvb.monitor(this.stop.id, payload.timeOffset, this.numberOfRequestedResults(payload)).then((data) => {
+            var response = {
+                id: payload.id,
+                connections: self.connectionsToBeDisplayed(data, payload)
+            };
+            self.sendSocketNotification("DVB-RESPONSE", response);
+        });
+    },
+    findStop: function(payload) {
+        var self = this;
+        Log.log("Searching stop ID for: " + payload.stopName);
+        dvb.findStop(payload.stopName).then((data) => {
+            if (Array.isArray(data) && data.length > 0) {
+                self.stop.id = data[0].id;
+                self.stop.name = payload.stopName;
+                self.monitor(payload);
+            }
+        });
+    },
     socketNotificationReceived: function(notification, payload) {
         if (notification === 'DVB-REQUEST') {
             var self = this;
-            dvb.findStop(payload.stopName).then((data) => {
-                if (Array.isArray(data) && data.length > 0) {
-                    var stopID = data[0].id;
-                    dvb.monitor(stopID, payload.timeOffset, this.numberOfRequestedResults(payload)).then((data) => {
-                        var response = {
-                            id: payload.id,
-                            connections: self.connectionsToBeDisplayed(data, payload)
-                        };
-                        self.sendSocketNotification("DVB-RESPONSE", response);
-                    });
-                }
-            });
+            if (!self.stop.id || self.stop.name != payload.stopName) {
```

Comment:
Could you please verify how the module behaves in case multiple instances are configured? As per my understanding NodeHelper is a single instance per module type, i.e. notifications from different modules with potentially different stop ids will be received. Since the stop variable in the NodeHelper contains a single value it will be flipping and, finally, no calls to dvb.findStop will be avoided.

Reply:
Good catch! Working on it.